pennylane/templates/decorator.py

```diff
@@ -0,0 +1,34 @@
+# Copyright 2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module contains the template decorator.
+"""
+from pennylane.utils import OperationRecorder
+
+def template(func):
+    """Wraps the given function and makes it return a list of all queued Operations.
```

Comment:
It would be good to have an example here, for users that end up on this docs page with no context. For example, just showing a minimal template with this decorator.

Also, it might be better to make the first line of the docstring something describing its purpose first, and move this more technical description down. How about the following for the first line?
```rest
Register a quantum template with PennyLane.
```
Vaguer, but communicates what it does/where it fits in the PL stack for less technical users.

Reply:
Done!